src/overlays/ovl_i6/fox_ve2.c

```diff
@@ -0,0 +1,4132 @@
+    D_800C9C30 = 0.0f;
+    boss->health = 100;
+    boss->fwork[18] = 1.0f;
+    boss->fwork[17] = 1.0f;
+    gLight1R = 0xFF;
+    gLight1G = 80;
+    gLight1B = 20;
+    gAmbientR = 1;
+    gAmbientG = 0;
+    gAmbientB = 20;
+    boss->timer_058 = 10;
+    boss->swork[4] = boss->swork[5] = 200;
+    boss->swork[6] = 100;
+    D_80178284 = 1;
+    D_80177AB0 = 1;
+    boss->info.hitbox[16] = 200.0f;
+    boss->info.hitbox[22] = 200.0f;
+    func_8001D444(0, 0x8021, 0, 0);
+}
+
+void func_i6_8018D2B0(Boss* boss) {
+    s32 i;
+    Vec3f sp68;
+
+    if ((boss->swork[5] < 0) && (boss->swork[4] < 0)) {
+        if ((boss->dmgType != 0) && (boss->dmgPart == 0 || boss->dmgPart == 1)) {
+            boss->dmgType = 0;
+            if (boss->actionState < 12) {
+                boss->health -= boss->damage;
+                boss->timer_05C = 20;
+                if ((boss->actionState == 9) || (boss->actionState == 10)) {
+                    if (boss->health <= 0) {
+                        D_Timer_80161A60 = 8;
+                        Audio_PlaySfx(0x2940D09A, &boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                        Audio_PlaySfx(0x31009063, &boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                        Audio_PlaySfx(0x19403070, &boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                        func_8001A55C(&boss->sfxPos, 0x11030073);
+                        func_800182F4(0x100100FF);
+                        func_800182F4(0x110100FF);
+                        boss->actionState = 31;
+                        boss->timer_050 = 200;
+                        gPlayer[0].state_1C8 = PLAYERSTATE_1C8_7;
+                        gPlayer[0].unk_1D0 = 0;
+                    } else if (boss->health < 50) {
+                        Audio_PlaySfx(0x2943500F, &boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                    } else {
+                        Audio_PlaySfx(0x29034003, &boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                    }
+                } else if (boss->health <= 0) {
+                    Audio_PlaySfx(0x19403070, &boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                    func_800182F4(0x100100FF);
+                    func_800182F4(0x110100FF);
+                    if (D_8017827C == 0) {
+                        boss->actionState = 30;
+                    } else {
+                        boss->actionState = 32;
+                    }
+                    boss->fwork[9] = 0.0f;
+                    boss->unk_04C = 0;
+                    D_80177A80 = 0;
+                    boss->timer_05C = 100;
+                } else {
+                    Audio_PlaySfx(0x29034003, &boss->sfxPos, 4, &D_800C5D34, &D_800C5D34, &D_800C5D3C);
+                }
+            }
+        }
+    } else if (boss->dmgType != 0) {
+        switch (boss->dmgPart) {
+            case 0:
+                if (boss->actionState < 11) {
```

Comment:
```suggestion
                if (boss->actionState <= 10) {
```
It looks like there is no action state 11. If I'm wrong, you can ignore this.

Reply:
yea, i have it get set to 11. 17 was highest i saw iirc